lib/bulk_processor/s3_file.rb

```diff
@@ -0,0 +1,83 @@
+require 'aws-sdk'
+
+class BulkProcessor
+  # Read and write files in a pre-configured S3 bucket.
+  class S3File
+    NAMESPACE = 'bulk_processor'.freeze
+    private_constant :NAMESPACE
+
+    # @param key [String] the unique identifier (within the bucket) used to
+    #   access the file
+    def initialize(key)
+      @key = "#{NAMESPACE}/#{key}"
+    end
+
+    def exists?
+      client.get_object(bucket: bucket, key: key)
+      true
+    rescue Aws::S3::Errors::NoSuchKey
+      false
+    end
+
+    # Yield the file stored in the bucket identified by the key. The file is
+    # only guaranteed to exist locally within the block, any attempts to access
+    # the file outside of the block will fail.
+    #
+    # @yields [File] a local copy of the remote file
```

Comment:
Ruby's `File#read` returns the contents of the file, so yielding the `File` is a bit surprising here. Maybe a different method name?